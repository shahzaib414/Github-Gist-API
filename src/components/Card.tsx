import React from 'react';
import styled from 'styled-components';

interface Props {
    children: React.ReactNode;
    title?: string
    onClick?: () => void
}

const CardView = styled.div`
    box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
    transition: 0.3s;
    display: flex;
    flex-direction: column;
    margin-top: 25px;
`;

const CardTitle = styled.div`
    margin: 20px;
    font-weight: bold;
`;

const CardContent = styled.div`
    padding: 20px;
    display: flex;
    flex-direction: column
`; 

const Card = ({ children, title, onClick }: Props) => {
    return <CardView>
        <CardTitle>
            {title}
        </CardTitle>
        <CardContent>{children}</CardContent>
    </CardView>
}

export default Card;