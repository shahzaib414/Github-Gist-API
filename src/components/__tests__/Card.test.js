import React from 'react';
import renderer from 'react-test-renderer';
import Card from "../Card";

test("Card Component renders correctly", () => {
    const component = renderer.create(<Card title="Card title">
        <h1> Card Children </h1>
    </Card>)
    expect(component).toMatchSnapshot();
})
