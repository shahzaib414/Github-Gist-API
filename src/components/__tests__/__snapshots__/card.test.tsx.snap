// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders correctly 1`] = `
<div
  className="sc-bdvvaa kCEU"
>
  <div
    className="sc-gsDJrp jKFXLn"
  >
    Card title
  </div>
  <div
    className="sc-dkPtyc gJUXbe"
  >
    <h1>
       Card Children 
    </h1>
  </div>
</div>
`;
